app/controllers/stores_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class StoresController < ApplicationController
+  schema(:directions) do
+    required(:address).value(:string)
+    optional(:transportation).value(:string)
+  end
+
+  def directions
+    user_coordinates_service = GetUserCoordinatesService.call(params:)
+
+    return if handle_service_error(user_coordinates_service)
+
+    directions_service = GetDirectionsService.call(
+      params: { transportation: params[:transportation], store_id: params[:id] },
+      user_coordinates: user_coordinates_service.result
+    )
+
+    return if handle_service_error(directions_service)
```

Comment:
should have method process_service to avoid DRY for calling multiple services, we only have 2 here, but what if there r 5
should you positive way to handle condition, rather use if error? than unless success?